download/upgradeRecipe/upgradeRecipe8.adoc

```diff
@@ -228,3 +273,722 @@ After in `*.xml`:
+
+Before in `*.java`:
+
+[source,java]
+----
+NaturalScoreComparator comparator = new NaturalScoreComparator();
+ConstraintMatchScoreComparator comparator2 = new ConstraintMatchScoreComparator();
+----
+
+After in `*.java`:
+
+[source,java]
+----
+Comparator<Score> comparator = Comparable::compareTo;
+Comparator<ConstraintMatch> comparator2 = Comparator.comparing(ConstraintMatch::getScore);
+----
+
+[.upgrade-recipe-minor.upgrade-recipe-public-api]
+=== `FeasibilityScore` removed
+
+The `FeasibilityScore` interface has been deprecated in 7.x and its only method `isFeasible()` moved to the `Score`
+supertype.
+The interface has now been removed.
+
+Users should refer to their ``Score``s by their ultimate type, eg. `HardSoftScore` as opposed to `Score`.
+
+[.upgrade-recipe-minor.upgrade-recipe-public-api]
+=== `@PlanningEntity.movableEntitySelectionFilter` removed
+
+The `movableEntitySelectionFilter` field on `@PlanningEntity` annotation has been deprecated in 7.x and a new field
+`pinningFilter` has been introduced, the name of which bears a clear relation to the `@PlanningPin` annotation.
+This filter implements a new `PinningFilter` interface, returning true if the entity is pinned, and false if movable.
+The logic of this new filter is therefore inverted as compared to the old filter.
+
+Users should update their `@PlanningEntity` annotations, supplying the new filter instead of the old.
+The old field has now been removed.
+
+Before in `*.java`:
+
+[source,java]
+----
+@PlanningEntity(movableEntitySelectionFilter = MyMovableEntitySelectionFilter.class)
+----
+
+After in `*.java`:
+
+[source,java]
+----
+@PlanningEntity(pinningFilter = MyPinningFilter.class)
+----
+
+[.upgrade-recipe-minor.upgrade-recipe-public-api]
+=== `@PlanningVariable.reinitializeVariableEntityFilter` removed
+
+The `reinitializeVariableEntityFilter` field on `@PlanningVariable` annotation has been deprecated for removal in 7.x and now removed.
+
+Users of this niche functionality should refer to the documentation on how to achieve the same result by
+link:https://docs.optaplanner.org/latest/optaplanner-docs/html_single/index.html#nullablePlanningVariable[power-tweaking construction heuristics].
+
+[.upgrade-recipe-minor.upgrade-recipe-public-api]
+=== `*ScoreHolder` classes turned into interfaces
+
+In OptaPlanner 7, `ScoreHolder` classes, used exclusively for
+https://docs.optaplanner.org/latest/optaplanner-docs/html_single/index.html#droolsScoreCalculation[Drools score calculation], exposed a number of public methods which, if used, allowed the user to unintentionally corrupt or otherwise negatively affect their scores.
+
+In OptaPlanner 8, these methods have been removed and the classes have been turned into interfaces.
+For the vast majority of users who did not use any of these potentially harmful methods, there will be no impact on their code.
+Others will find suitable replacements in the public API in areas of
```

Comment:
writing style: don't talk over their head, as if they're children, but make it more direct:
"For the vast majority of users who did not use any of these ..." -> "You probably don't use any of the removed, potentially harmful methods, so ..."
These is a good example where the "avoid "you"" rule doesn't apply.